### PR TITLE
api.info

### LINKED ISSUE #34: [feature] api.info.* - Way to communicate with user from lua rule/implementation
We should allow (rule/implementation)makers to communicate with user in order to e.g. log information, share info why rule does not pass, share errors etc.
I suggest to implement following functions:
- `api.info.log`
- `api.info.debug`
- `api.info.error`
- `api.info.warn`
- `api.info.important` (here I am not sure whether it is necessary)

Keep in mind that e.g. `api.info.log` should not directly print content into terminal, because in the future we're planning to make GUI for our app and then api.info.log should handle GUI instead of this CLI.
I think we can resolve it by putting logic responsible directly for printing content into `src/cmd` dir

### PR DESCRIPTION
**Pull request by BaderBC**
Closes #34

- implemented api.info for cli
- implemented api.info for gui
- cmd flag and guiApi fix

### CODE REVIEW

File: src/cmd/root.go [hunk 1]
```diff
@@ -22,5 +22,9 @@ func Execute() {
 }
 
 func init() {
+	rootCmd.AddCommand(checkFileCmd)
 	rootCmd.AddCommand(checkCmd)
+
+	checkFileCmd.Flags().Bool("gui-child-mode", false, "Tells app that it is executed by gui")
```
Comment (cooligus) [inline src/cmd/root.go:28]: I would consider creating variables such as `commandName`, `commandDesc` and `commandDefOption`

File: src/cmd/check.go [hunk 2]
```diff
@@ -2,28 +2,84 @@ package cmd
...
 
+var checkFileCmd = &cobra.Command{
+	Use:   "check file {path}",
+	Short: "Check local lua rule file",
+	Args:  cobra.ExactArgs(2),
+	Run: func(cmd *cobra.Command, args []string) {
+		runtimeData := getInitialRuntimeData(cmd)
+		script, err := os.ReadFile(args[1])
+		if err != nil {
+			panic(err)
```
Comment (cooligus) [inline src/cmd/check.go:22]: I don't trust it

**Review by cooligus — CHANGES_REQUESTED** [review]

File: src/cmd/root.go (hunk 1 above)
Comment (BaderBC) [reply src/cmd/root.go:28]: I don't think it is necessary to create const variable only for holding flag name, default value and description especially that these variables in such case would be used each only once.

**Review by sewe2000 — APPROVED** [review]

File: src/cmd/check.go (hunk 2 above)
Comment (BaderBC) [reply src/cmd/check.go]: I change my mind about this particular panic and resolved it in the newest commit

**Review by cooligus — APPROVED** [review]

**Pull request merged**